ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -69,6 +70,23 @@ static int gcp_object_list_free(WT_FILE_SYSTEM *, WT_SESSION *, char **, uint32_
   __attribute__((__unused__));
 static int gcp_terminate(WT_STORAGE_SOURCE *, WT_SESSION *) __attribute__((__unused__));
 
+// Construct a pathname from the directory and the object name.
+static std::string
+gcp_path(const std::string &dir, const std::string &name)
+{
+    // Skip over "./" and variations (".//", ".///./././//") at the beginning of the name.
+    int i = 0;
+    while (name[i] == '.') {
+        if (name[i + 1] != '/')
+            break;
+        i += 2;
+        while (name[i] == '/')
+            i++;
+    }
```

Comment:
I would use the function find_first_not_of here, https://cplusplus.com/reference/string/string/find_first_not_of/ instead of the loop.